hydra-explorer/test/Hydra/Explorer/ExplorerStateSpec.hs

```diff
@@ -16,15 +16,25 @@ spec = do
     -- Even if we only observe a part of the life cycle of some head.
     prop "Any head observations (of some head id) must yield an entry of that head id" $
       forAll genObservations $ \observations ->
-        aggregateHeadObservations observations [] =/= []
+        let ExplorerState{heads} = aggregateHeadObservations observations (ExplorerState [] initialTickState)
+         in heads =/= []
     prop "Given any observations, the resulting list of head ids is a prefix of the original" $
       forAll genObservations $ \observations ->
-        forAll arbitrary $ \initialState -> do
-          let resultHeads = aggregateHeadObservations observations initialState
-          getHeadIds initialState `isPrefixOf` getHeadIds resultHeads
+        forAll arbitrary $ \initialHeads -> do
+          let resultExplorerState = aggregateHeadObservations observations (ExplorerState initialHeads initialTickState)
+          getHeadIds initialHeads `isPrefixOf` getHeadIds (heads resultExplorerState)
  where
-  genObservations :: Gen [HeadObservation]
-  genObservations = arbitrary `suchThat` (not . null) `suchThat` notElem NoHeadTx
+  genObservations :: Gen [ChainObservation]
+  genObservations =
+    arbitrary
+      `suchThat` (not . null)
+      `suchThat` ( not
+                    . any
+                      ( \case
+                          HeadObservation{} -> False
+                          Tick{} -> True
+                      )
+                 )
```

Comment:
Should: avoid `suchThat` as it makes inefficient generators.

An alternative implementation could be something like:

```suggestion
  genObservations :: Gen [ChainObservation]
  genObservations = 
    listOf1 $ HeadObservation <$> arbitrary <*> arbitrary <*> arbitrary

```